jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -8307,13 +8307,71 @@ public String resolveConstantDesc(MethodHandles.Lookup lookup) {
 	}
 	
 	/**
-	 * Apply a function to this string. The function expects a single String input and returns an R.
+	 * Apply a function to this string. The function expects a single String input
+	 * and returns an R.
 	 * 
-	 * @param f - the functional interface to be applied
+	 * @param f 
+	 *			  the functional interface to be applied
+	 *
 	 * @return the result of application of the function to this string
+	 * 
+	 * @since 12
 	 */
 	public <R> R transform(Function<? super String, ? extends R> f) {
 		return f.apply(this);
 	}
+
+	/**
+	 * Indents each line of the string depending on the value of n, and normalizes
+	 * line terminators to the newline character "\n".
+	 * 
+	 * @param n 
+	 *			  the number of spaces to indent the string
+	 *
+	 * @return the indented string with normalized line terminators
+	 * 
+	 * @since 12
+	 */
+	public String indent(int n) {
+		Stream<String> lines = lines();
+		Iterator<String> iter = lines.iterator();
+		StringBuilder builder = new StringBuilder();
+
+		while (iter.hasNext()) {
+			String currentLine = iter.next();
+
+			if (n > 0) {
+				int spaceOffset = n + currentLine.length();
+				currentLine = String.format("%" + spaceOffset + "s", currentLine);
+			} else if (n < 0) {
+				int absN = Math.abs(n);
+				if (currentLine.isBlank()) {
+					if (currentLine.length() > absN) {
+						currentLine = currentLine.substring(absN);
+					} else {
+						currentLine = "";
+					}
+				} else {
+					int start = 0;
+					while ((currentLine.length() > start) 
+						&& (Character.isWhitespace(currentLine.charAt(start)))
+					) {
+						start++;
+						if (start >= absN) {
+							break;
+						}
+					}
+					currentLine = currentLine.substring(start);
+				}
+			}
+			/**
+			 * Newline characters are removed when lines() is called. The newline character
+			 * "\n" is added back for each line, to normalize line terminators.
+			 */
+			builder.append(currentLine + "\n");
```

Comment:
What's the expected behaviour on windows?  Appending `\n`?  or `\r\n`?  I expect using `System.lineSeparator()` will do the right thing here

Reply:
Looks like the jshell output for RI on Windows doesn't use `System.lineSeparator()` and just uses `\n`. Will be changing it back to `\n`.